provisional X100F support

dtorop opened the pull request:
Crop (and CFA) are derived from sample images. The sensor black/white are copied from Fujifilm X-T2 and appear workable, but are not derived from a DNG.

dtorop committed:
cameras.xml: provisional support for X100F

This is copied from Fujifilm X-T2.

TODO: check crop and black/white points

dtorop committed:
cameras.xml: X100F crop adjustment

Set crop based on experimentation with a sample image.

TODO: black/white points

LebedevRI commented on data/cameras.xml:
```diff
@@ -8520,6 +8520,19 @@
+			<ColorRow y="5">GGRGGB</ColorRow>
+		</CFA2>
+		<Crop x="0" y="0" width="-126" height="0"/>
+		<Sensor black="1024" white="16383"/>
```
@dtorop 
Re black/white points: all the decoders need to be extended to read all of this info from the raw files themselves.
cameras.xml 
E.g. In this case: https://raw.pixls.us/getfile.php/1166/exiftool/DSCF0150.RAF.exiftool.txt
```
[RAF]           XTransLayout                    : BGGRGG RGGBGG GBRGRB RGGBGG BGGRGG GRBGBR
[RAF]           BlackLevel                      : 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022 1022
```
I'd guess the CFA is specified for crop
```
[MakerNotes]    RawImageWidth                   : 6032
[MakerNotes]    RawImageHeight                  : 4032
```
while the full image is
```
[RAF]           RawImageFullWidth               : 6160
[RAF]           RawImageFullHeight              : 4032
```

dtorop committed:
cameras.xml: revised Fujifilm X100F values

Set blackpoint as found in example
https://raw.pixls.us/getfile.php/1166/exiftool/DSCF0150.RAF.exiftool.txt. Narrow
crop by 2 pixels to make output image 6032x4032, to match
RawImageWidth as found in that EXIF. Note that this crops two columns
of data which do seem to record image data, not black.

LebedevRI commented on data/cameras.xml:
```diff
@@ -8530,8 +8530,8 @@
 		</CFA2>
-		<Crop x="0" y="0" width="-126" height="0"/>
-		<Sensor black="1024" white="16383"/>
+		<Crop x="0" y="0" width="-128" height="0"/>
```
> Narrow crop by 2 pixels to make output image 6032x4032, to match RawImageWidth as found in that EXIF.

No point in doing that, as per https://github.com/darktable-org/rawspeed/blob/develop/README.md

```
RawSpeed does NOT…
* crop the image to the same sizes as manufactures, but supplies biggest possible images.
```

dtorop committed:
cameras.xml: restore wider X100F crop

As per README, no need to match manufacturer image size.

LebedevRI commented:
@dtorop, @schenlap: thank you both!
